iroh-relay/src/server.rs

```diff
@@ -140,12 +140,41 @@ pub struct TlsConfig<EC: fmt::Debug, EA: fmt::Debug = EC> {
 }
 
 /// Rate limits.
+// TODO: accept_conn_limit and accept_conn_burst are not currently implemented.
 #[derive(Debug, Default)]
 pub struct Limits {
     /// Rate limit for accepting new connection. Unlimited if not set.
     pub accept_conn_limit: Option<f64>,
     /// Burst limit for accepting new connection. Unlimited if not set.
     pub accept_conn_burst: Option<usize>,
+    /// Rate limits for incoming traffic from a client connection.
+    pub client_rx: ClientConnRateLimit,
+}
+
+/// Per-client rate limit configuration.
+#[derive(Debug, Copy, Clone)]
+pub struct ClientConnRateLimit {
+    /// Max number of bytes per second to read from the client connection.
+    ///
+    /// Defaults to [`NonZeroU32::MAX`], effectively unlimited.
+    pub bytes_per_second: NonZeroU32,
+    /// Max number of bytes to read in a single burst.
+    ///
+    /// Defaults to [`NonZeroU32::MAX`], effectively unlimited.
+    pub max_burst_bytes: NonZeroU32,
+}
+
+impl ClientConnRateLimit {
+    pub(super) const MAX: ClientConnRateLimit = ClientConnRateLimit {
+        bytes_per_second: NonZeroU32::MAX,
```

Comment:
To be fair this limits it to 4GB/sec, while not low, definitely not unlimited.

Reply:
Yeah, I did this for simplicity.  Do you think this is unreasonable and we should do truly unlimited?

Reply:
So unfortunately the rate limiter doesn't support u64 and I didn't look into why.  I assume they have a reason.

And using 0 for unlimited is both a type-level sin and also a problem for max_burst_bytes since 0 is actually a meaningful value and would be ambiguous and very confusing to users.